Update values associated with "Low" and "Medium" image quality settings

SiobhyB opened the pull request:
> **Note**
> Although this PR is large, the vast majority of file changes are the result of a change to a string name, as can be viewed in https://github.com/wordpress-mobile/WordPress-Android/pull/19644/commits/452ed672ac3d147571376dc954eb907bb4f34c23.

Merging into https://github.com/wordpress-mobile/WordPress-Android/pull/19581.

## Description

This PR changes the default values associated with the `Low` and `Medium` image quality settings found under _Me_ → _App Settings_.

The `Low` value has been changed from `80` to `70`, and the `Medium` value has been changed from `85` to `80`. This brings the Android app with the values introduced for iOS in https://github.com/wordpress-mobile/WordPress-iOS/pull/21981.

We've decided to change these values without explicitly notifying users, with the following in mind:

- Users don’t technically select `85`. They choose `Medium` or `Low`. The decision around what these values meant was set around 6 or 7 years ago. We think it’s fair to reassess what `Medium` or `Low` means in this day and age (now that image quality tends to be a lot higher, and therefore 80% compression may not be as visually impactful as it would have been years ago).
- 5% isn’t that meaningful a difference, and we wouldn't expect most users who've chosen `Medium` to notice it.
- Those users who've selected `Low` know the route to follow to change this setting, and have already manually indicated that they value speed over quality.

Related discussion: pcdRpT-4tj-p2 

-----

## To Test

- Add an image to your phone's library with a large file size.
- Navigate to _Me_ → _App Settings_ and ensure **Optimize Images** is toggled on.
- Select either **Low** or **Medium** from the **Image Quality** dropdown.
- Navigate to the post editor and add an image block.
- Tap on the **Choose from device** option.
- Select the image previously added with a large file size, and wait for the image upload to finish.
- Go to the browser and navigate to the **Media** screen of the site.
- Select the uploaded image and click on the **Edit** button.
- Observe that the file size under the FILE SIZE section is smaller than the original file size and is of the quality you'd expect.

-----

## Regression Notes

1. Potential unintended areas of impact

    - As we're changing numerous language files, there's a chance translations could be unintentionally impacted.

2. What I did to test those areas of impact (or what existing automated tests I relied on)

    - I manually tested the app in a few different languages to verify there were no regressions.

3. What automated tests I added (or what prevented me from doing so)

    - Automated tests for the image optimization settings are to follow in a separate PR.

-----

## PR Submission Checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

SiobhyB committed:
Update values for low and medium image quality

To match the default values for image quality for iOS (see: https://github.com/wordpress-mobile/WordPress-iOS/pull/21981), this commit changes the defaults for quality to the following:

- Low = 70%
- Medium = 80%

SiobhyB committed:
Update string names for accuracy

Following the last commit, the numbers in the string names no longer reflect reality. As such, they're updated in this commit to reflect the level of quality they represent.

fluiddot commented on WordPress/src/main/res/values/key_strings.xml:
```diff
@@ -356,16 +356,16 @@
         <item>80</item>
-        <item>85</item>
         <item>90</item>
         <item>95</item>
```
I'd like to note that the _Very High_ value wasn't added in the iOS version: https://github.com/wordpress-mobile/WordPress-iOS/pull/21981/files#diff-8dba2169bbf9087083db06c81198cd41a1b6a09b33f224fbed73f47dac842b15R22-R39.

fluiddot replied:
Seems we'd need to keep it as users on Android might have this preference set in previous versions.

SiobhyB replied:
I think it'd be great to simplify this to match iOS, but agree it's trickier to do now that the setting has been out there for so long for Android users. I don't think we can really take it away.

fluiddot commented on WordPress/src/main/res/values/key_strings.xml:
```diff
@@ -356,16 +356,16 @@
     <string-array name="site_settings_image_quality_values" translatable="false">
+        <item>70</item>
         <item>80</item>
-        <item>85</item>
```
I think on Android we use these values to persist the value of the _Image Quality_ setting. Removing the value `85` leads to an inconsistency in this setting if it was previously set and the app is updated with these changes.

**Steps:**
- Install a build previous to these changes ([jetpack-prototype-build-pr19581-eb71010.apk](https://d2twmm2nzpx3bg.cloudfront.net/8204deb9481772038e97669f4e11f5a5e6b4f94f/jetpack-prototype-build-pr19581-eb71010.apk)).
- Open the app and go to Me -> App Settings -> Image Quality.
- Choose _Medium_ option.
- Install the build with these changes ([jetpack-prototype-build-pr19644-452ed67.apk](https://d2twmm2nzpx3bg.cloudfront.net/8204deb9481772038e97669f4e11f5a5e6b4f94f/jetpack-prototype-build-pr19644-452ed67.apk)).
- Open the app and go to Me -> App Settings -> Image Quality.
- Observe that the value set in _Image Quality_ is _Low_. 
- Tap on _Image Quality_ and observe that no quality option is checked.

As far as I tested, the process of uploading an image is unaffected. However, I noticed that the quality used is the one originally set (85%) instead of the new one (80%). Regarding the _Low_ quality label shown in App Settings, I presume it's showing it because it defaults to the first item of the list (i.e. Low) when it can't fetch the proper value.

We might consider automatically updating the setting for users affected by this case to the new value for Medium Quality. WDYT?

SiobhyB replied:
Thank you for spotting this issue, @fluiddot! I made a change that addresses it in https://github.com/wordpress-mobile/WordPress-Android/pull/19644/commits/7109245deee9ac10a906173971dfa876cc031510. As part of that change, there's now a check that ensures the quality value exists in the array. If it doesn't exist, the default value is set.

I'm thinking this approach will help us if we want to make further changes in the future.

Let me know what you think or if you perhaps see a different approach to take here.

fluiddot replied:
Confirmed this is solved in recent changes (https://github.com/wordpress-mobile/WordPress-Android/pull/19644/commits/7109245deee9ac10a906173971dfa876cc031510).

SiobhyB committed:
Merge branch 'feature/updates-to-optimization-settings' into feature/update-optimization-values

SiobhyB committed:
Add check to ensure quality value is valid

Addresses bug reported here: https://github.com/wordpress-mobile/WordPress-Android/pull/19644/files#r1402179477

As we've made changes to the values in the quality array, there may be times when users have previously chosen values that no longer exist. Those cases were not being properly handled in the code.

With this commit, we now check if a quality value exists in the array. If it doesn't, we return the default quality value.

SiobhyB committed:
Import Context for use in class

fluiddot commented on WordPress/src/main/java/org/wordpress/android/ui/prefs/AppPrefs.java:
```diff
@@ -777,7 +778,19 @@ public static void setImageOptimizeQuality(int quality) {
+
+        // If quality int does not exist in settings array, return the default quality value instead
+        if (!isQualityValid) {
+            setImageOptimizeQuality(WPMediaUtils.OPTIMIZE_IMAGE_ENCODER_QUALITY);
```
In the case we previously persisted an invalid quality value (e.g. 85), since we are using the default value, I think we should explicitly return it. Otherwise, it will return the invalid `quality` value, which leads to the following state when tapping on _Image Quality_ option:

<img width="300" alt="Screenshot 2023-11-23 at 11 26 25" src="https://github.com/wordpress-mobile/WordPress-Android/assets/14905380/cab982b7-fac2-42e3-9655-8dd63ac11b80">

fluiddot replied:
```suggestion
            setImageOptimizeQuality(WPMediaUtils.OPTIMIZE_IMAGE_ENCODER_QUALITY);
            return WPMediaUtils.OPTIMIZE_IMAGE_ENCODER_QUALITY;
```

SiobhyB replied:
Aha, good catch! 🤦‍♀️  

I've made sure we're returning the default value in https://github.com/wordpress-mobile/WordPress-Android/pull/19644/commits/acb9197254838ff873477d972cd5f1a214c0ef85.

SiobhyB committed:
Ensure we return default quality int if necessary

In cases when a quality value is invalid, we should explicitly detect that and then return the default quality value. Otherwise, the invalid value will continue to be returned.

Response to the following feedback: https://github.com/wordpress-mobile/WordPress-Android/pull/19644#discussion_r1403179770

fluiddot reviewed (approved):
LGTM 🎊 !
